From 3006d06fe462fc2a18e872b8b69001d8a74bbb93 Mon Sep 17 00:00:00 2001
Message-Id: <3006d06fe462fc2a18e872b8b69001d8a74bbb93.1566848730.git.Jayanth_Raghuram@Dell.com>
From: Jayanth Raghuram <Jayanth_Raghuram@Dell.com>
Date: Mon, 26 Aug 2019 13:33:52 -0500
Subject: [PATCH] MdePkg/DxeHstiLib: Added checks to improve error handling.
 Added checks for return parameters of LocateHandleBuffer & GetSupportedTypes
 function calls in InternalHstiFindAip to improve error handling. An issue was
 observed on Dell Poweredge R740, where the Dell PERC H740P controller UEFI
 driver returned InfoTypesBuffer = NULL, InfoTypesBufferCount = 0 and caused
 an FreePool assert.

Signed-off-by: Jayanth Raghuram <Jayanth.Raghuram@Dell.com>
Cc: Wei G Liu <Wei_G_Liu@Dell.com>
---
 MdePkg/Library/DxeHstiLib/HstiDxe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/MdePkg/Library/DxeHstiLib/HstiDxe.c b/MdePkg/Library/DxeHstiLib/HstiDxe.c
index 9466e3dcb3..104e9d4aab 100644
--- a/MdePkg/Library/DxeHstiLib/HstiDxe.c
+++ b/MdePkg/Library/DxeHstiLib/HstiDxe.c
@@ -51,7 +51,7 @@ InternalHstiFindAip (
                   &NoHandles,
                   &Handles
                   );
-  if (EFI_ERROR (Status)) {
+  if (EFI_ERROR (Status) || (Handles == NULL) || (NoHandles == 0)) {
     return NULL;
   }
 
@@ -77,7 +77,7 @@ InternalHstiFindAip (
                     &InfoTypesBuffer,
                     &InfoTypesBufferCount
                     );
-    if (EFI_ERROR (Status)) {
+    if (EFI_ERROR (Status) || (InfoTypesBuffer == NULL) || (InfoTypesBufferCount == 0)) {
       continue;
     }
 
-- 
2.19.1.windows.1

